lib/immutable-struct.rb

```diff
@@ -124,10 +124,17 @@ def self.new(*attributes,&block)
     klass.class_exec(imethods) do |imethods|
       define_method(:to_h) do
         imethods.inject({}) do |hash, method|
-          next hash if [:==, :eql?, :merge, :hash].include?(method)
+          next hash if [:to_json, :==, :eql?, :merge, :hash].include?(method)
           hash.merge(method.to_sym => self.send(method))
         end
       end
+
+      define_method(:to_json) do |*args|
+        imethods.inject({}) do |hash, method|
+          next hash if [:to_json, :to_hash, :to_h, :==, :eql?, :merge, :hash].include?(method)
```

Comment:
The method list here looks similar to the method list in the `to_h` method.  It's not so obvious why they are different.  Would it be helpful to assign those lists to constants with good names and/or add a comment?